src/decode.rs

```diff
@@ -4994,7 +4994,10 @@ pub fn rav1d_submit_frame(c: &Rav1dContext, state: &mut Rav1dState) -> Rav1dResu
     ) {
         fc.task_thread.error.store(1, Ordering::Relaxed);
         let _ = mem::take(&mut *fc.in_cdf.try_write().unwrap());
-        if f.frame_hdr.as_ref().unwrap().refresh_context != 0 {
+        if f.frame_hdr
+            .as_ref()
+            .is_some_and(|frame_hdr| frame_hdr.refresh_context != 0)
+        {
```

Comment:
I'm not sure what the right behavior is in the `None` case, but crashing is certainly not it

Reply:
Places where there is an unwrap should have been unchecked derefs in C, so let me check what C was doing here.

Reply:
So C is doing the same thing as we are here: 
* https://github.com/memorysafety/rav1d/blob/c7d127e7e31bd3366ac7dc1717dda9782905c605/src/decode.c#L3350
* https://code.videolan.org/videolan/dav1d/-/blob/master/src/decode.c?ref_type=heads#L3722

Can you check if you get an error/crash on `dav1d` here or not?  Thanks!

Reply:
Using https://crates.io/crates/dav1d 0.10.3 I do not get a crash, but I get an `Invalid argument` error and some useful info printed:

```
Error parsing frame header
Error parsing OBU data
```

It should be noted though that dereferencing a null pointer is not necessarily a crash in C - it is UB. The compiler is free to assume the pointer is not null, and do whatever crazy shenanigans that implies.

Reply:
Hmm, it's possible UB could be doing something weird here, but I'd still expect a segfault for the most part.  We likely diverged at some point, so we'll have to figure out where and see if that's something we need to fix or if it's okay to crash, just in a different place.

That said, this video should successfully decode with normal `dav1d`, but doesn't?  Is that a bug with `dav1d` as well?

Reply:
I don't know if the video should successfully decode, or if it is actually a bad video.

> or if it's okay to crash

I would prefer a `rav1d` that never crashes on any input, but returns errors.

Reply:
> I would prefer a rav1d that never crashes on any input, but returns errors.

Yup, that would definitely be preferable.